Assets/Editor/AddScenesToBuild.cs

```diff
@@ -0,0 +1,90 @@
+using System.Collections.Generic;
+using UnityEditor;
+using UnityEditor.SceneManagement;
+using UnityEngine;
+
+[InitializeOnLoad]
+public class AddScenesToBuild : EditorWindow
+{
+    [MenuItem("QA Tools/Open Core Scene Menu")]
+    static void OpenScene()
+    {
+        EditorSceneManager.OpenScene("Assets/QA/Tests/Core Platform Menu/Core Platforms Menu.unity");
+    }
+
+    [MenuItem("QA Tools/Add All Core Samples to Build")]
+    private static void AddAllScenesToBuildExcludingXboxAndXR()
+    {
+        // Get all available scenes in the project
+        string[] sceneGuids = AssetDatabase.FindAssets("t:Scene");
+        string[] scenePaths = new string[sceneGuids.Length];
+
+        for (int i = 0; i < sceneGuids.Length; i++)
+        {
+            scenePaths[i] = AssetDatabase.GUIDToAssetPath(sceneGuids[i]);
+        }
+        // Filter out scenes in folders containing "xbox" or "xr"
+        List<string> filteredScenePaths = new List<string>();
+        foreach (string path in scenePaths)
+        {
+            if (!IsPathInExcludedFolder(path))
+            {
+                filteredScenePaths.Add(path);
+            }
+        }
+
+        // Ensure "Core Platforms Menu" is at the beginning of the list
+        filteredScenePaths.Insert(0, "Assets/QA/Tests/Core Platform Menu/Core Platforms Menu.unity");
```

Comment:
I was wondering if this gets added twice as scenePaths would pickup the scene too ?

Reply:
agreed, would recommend to either skip/filter it out in the foreach above and add it like this or detect in within the loop and use Insert(0 instead of the Add for that particular file.